adr/2025-06-03-integrating-the-language-pack-into-platform.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Integrating the language pack into platform
+date: 2025-06-03
+area: discovery
+tags: [plugin, languages, language-pack, translations, crowdin]
+---
+
+## Context
+The Shopware Language Pack plugin enables us to distribute translations from Crowdin to Shopware installations. While this was a convenient solution in the past, while most of our repositories and workflows were kept private, this approach is unnecessarily convoluted and cumbersome for developers and users in that changes to any snippet require multiple steps:
+
+1. Translations are updated in CrowdIn (by the community or shopware)
+2. Merging the changes from CrowdIn into `shopware/translations` (automated, may require manual review)
+3. Merging the changes from `shopware/translations` into `shopware/SwagLanguagePack`, which primarily just distributes json files (automated, may require manual review)
+4. Publishing Language Pack releases the store (automated, may require manual input)
+5. Updating Language Pack installation
+   - OnPrem: Manually updating the plugin in a Shopware installation
+   - Cloud: Updating the docker image dependency
+
+This sequence also introduces overhead in maintenance and CI/CD
+
+### Background and Motivation
+* The current setup was influenced by limitations of our previous GitLab-based workflow, where distribution via plugin was the most practical method. This workflow has been in place for ~6 years.
+* Crowdin remains our source of truth for all supported language snippets, and community members can contribute translations directly.
+* `shopware/translations` acts as an intermediary data source, so we are not coupled to Crowdin in core Shopware repositories.
+* The primary goal of this change is to reduce maintenance effort by replacing steps 3–5 with a single step, and removing `shopware/SwagLanguagePack` from the workflow entirely. The new step should be in `shopware/shopware`.
+
+## Decision
+We will implement a new service in `shopware/shopware` (ie as part of the Shopware platform) to download translations right from their [GitHub Repository](https://github.com/shopware/translations/) and manage them without the need to install any extension.
+Translations will be downloaded as JSON files (via admin user interaction or command execution) and stored on the local file system, just like existing platform snippet files.
```

Comment:
only local? what about other used file systems?

Reply:
Because of cloud? In cloud, the snippets will be downloaded once for all instances.